src/mlpack/core/cv/k_fold_cv.hpp

```diff
@@ -280,30 +281,42 @@ class KFoldCV
   /**
    * Get the ith training subset from a variable of a matrix type.
    */
-  template<typename ElementType>
-  inline arma::Mat<ElementType> GetTrainingSubset(arma::Mat<ElementType>& m,
-                                                  const size_t i);
+  template<typename SubsetMatType>
+  inline SubsetMatType GetTrainingSubset(SubsetMatType& m,
+                                         const size_t i,
+                                         const typename std::enable_if_t<
+                                           IsMatrix<SubsetMatType>::value
+                                         >* = 0);
```

Comment:
```suggestion
  inline SubsetMatType GetTrainingSubset(
      SubsetMatType& m,
      const size_t i,
      const typename std::enable_if_t<IsMatrix<SubsetMatType>::value>* = 0);
```
Just a suggestion, I feel like this is easier to read (and similar for the other overloads).